cyber/record/record_writer.h

```diff
@@ -35,48 +35,134 @@ namespace apollo {
   using MessageProtoDescMap = std::unordered_map<std::string, std::string>;
   using FileWriterPtr = std::unique_ptr<RecordFileWriter>;
 
+  /**
+  * @brief The default constructor.
+  */
   RecordWriter();
+
+  /**
+  * @brief The constructor with record header as parameter.
+  *
+  * @param header
+  */
   explicit RecordWriter(const proto::Header& header);
 
+  /**
+  * @brief Virtual Destructor.
+  */
   virtual ~RecordWriter();
 
+  /**
+  * @brief Open a record to write.
+  *
+  * @param file
+  *
+  * @return True for success, false for fail.
+  */
   bool Open(const std::string& file);
 
+  /**
+  * @brief Clean the record.
+  */
   void Close();
 
+  /**
+  * @brief Write a channel to record.
+  *
+  * @param channel_name
+  * @param message_type
+  * @param proto_desc
+  *
+  * @return True for success, false for fail.
+  */
   bool WriteChannel(const std::string& channel_name,
                     const std::string& message_type,
                     const std::string& proto_desc);
 
+  /**
+  * @brief Write a message to record.
+  *
+  * @tparam MessageT
+  * @param channel_name
+  * @param message
+  * @param time_nanosec
+  * @param proto_desc
+  *
+  * @return True for success, false for fail.
+  */
   template <typename MessageT>
   bool WriteMessage(const std::string& channel_name, const MessageT& message,
                     const uint64_t time_nanosec,
                     const std::string& proto_desc = "");
 
+  /**
+  * @brief Set max size (KB) to segement record file
+  *
+  * @param size_kilobytes
+  *
+  * @return True for success, false for fail.
+  */
   bool SetSizeOfFileSegmentation(uint64_t size_kilobytes);
 
+  /**
+  * @brief Set max interval (Second) to segement record file.
+  *
+  * @param time_sec
+  *
+  * @return True for success, false for fail.
+  */
   bool SetIntervalOfFileSegmentation(uint64_t time_sec);
 
+  /**
+  * @brief Get message number by channel name.
+  *
+  * @param channel_name
+  *
+  * @return Message number.
+  */
   uint64_t GetMessageNumber(const std::string& channel_name) const override;
 
+  /**
+  * @brief Get message type by channel name.
+  *
+  * @param channel_name
+  *
+  * @return Message type.
+  */
   const std::string& GetMessageType(
       const std::string& channel_name) const override;
 
+  /**
+  * @brief Get proto descriptor string by channel name.
+  *
+  * @param channel_name
+  *
+  * @return Proto descriptor string by channel name.
+  */
   const std::string& GetProtoDesc(
       const std::string& channel_name) const override;
 
-  bool IsNewChannel(const std::string& channel_name) const;
+  /**
+  * @brief Get channel list.
+  *
+  * @return List container with all channel name string.
+  */
+  std::set<std::string> GetChannelList() const override;
 
  private:
   bool WriteMessage(const proto::SingleMessage& single_msg);
   bool SplitOutfile();
+  bool IsNewChannel(const std::string& channel_name) const;
```

Comment:
why bring back to be private?

Reply:
Since it has already checked whether is new channel in WriteChannel().
Anyway, I will reconfirm the change and push a new one.

Reply:
yea but it will print a warning log if existing inside of WriteChannel, which is really noisy if write large number of messages

Reply:
OK，I move this method to public.